# Lab 3

Mauricio Garcia 11/14/2019

### Abstract

In this lab, I mess around with a crime stats dataset and perform some mathematical calculations on it. The purpose of this lab overall was to see what crime data would look like from a couple different perspectives as they span over time. The data collection was based in Seattle.

### Dataset preparation

I import all the modules I need for this lab below.

In [30]:
# Here are some modules
import pandas as pd
import numpy as np
import statistics
import scipy as sp
import matplotlib.pyplot as plt
import seaborn as sns

### Data modeling

In [31]:
# And I will open up my file here
df = pd.read_csv('crimenes.csv')
# Here, I show the first couple rows
df.head(5)

Report_Year  Census_Tract_1990               Crime_Type  Report_Year_Total
0         1996                1.0       Aggravated Assault                 11
1         1996                1.0                 Homicide                  0
2         1996                1.0  NonResidential Burglary                 41
3         1996                1.0    Property Crimes Total                430
4         1996                1.0                     Rape                  2

In [32]:
# The dataset looks pretty clean so I can go ahead and open up more of the file. I want to now look at the end of this dataset.
df.tail(5)

Report_Year  Census_Tract_1990            Crime_Type  Report_Year_Total
14263         2007               99.0                  Rape                  4
14264         2007               99.0  Residential Burglary                 36
14265         2007               99.0               Robbery                  3
14266         2007               99.0                 Theft                147
14267         2007               99.0         Vehicle Theft                 52

In [33]:
# I think it's important to also take a look at general info on this dataset, so here's just that
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14268 entries, 0 to 14267
Data columns (total 4 columns):
Report_Year          14268 non-null int64
Census_Tract_1990    14236 non-null float64
Crime_Type           14268 non-null object
Report_Year_Total    14268 non-null int64
dtypes: float64(1), int64(2), object(1)
memory usage: 446.0+ KB


As you can see, there are a total of 14,268 entries. Now, I am ready to analyze this data.

### Data analysis

Here, I will perform a series of mathematical calculations to gather more information from the data that may not stick out as much from simply glancing at the data.

In [34]:
# I want to find the mean of my columns, so for that, I'll use this code
df.mean()

Report_Year          2001.327586
Census_Tract_1990      60.955128
Report_Year_Total      75.137090
dtype: float64

Out of the things here, one of the rows of information is actually usable. The average of the report years and the census tract doesn't really matter much to me, but I find it most interesting to see how many crimes would be logged on average every year

In [35]:
# now let's organize the dataset differently and look at the criminal cases 
df.sort_values("Crime_Type")

Report_Year  Census_Tract_1990            Crime_Type  Report_Year_Total
0             1996               1.00    Aggravated Assault                 11
1350          1997              11.00    Aggravated Assault                  4
8070          2002              52.00    Aggravated Assault                 26
8060          2002              51.00    Aggravated Assault                  7
11790         2005              54.00    Aggravated Assault                 12
3480          1998              80.00    Aggravated Assault                103
8050          2002              50.00    Aggravated Assault                  0
1340          1997             109.00    Aggravated Assault                 35
11800         2005              55.00    Aggravated Assault                  1
8040          2002               5.00    Aggravated Assault                  1
3490          1998              81.00    Aggravated Assault                215
11810         2005              56.00    Aggravated Assault                  2
8030          2002              49.00    Aggravated Assault                 15
8020          2002              48.00    Aggravated Assault                  7
1330          1997             108.00    Aggravated Assault                 14
3500          1998              82.00    Aggravated Assault                 20
11820         2005              57.00    Aggravated Assault                  4
8010          2002              47.00    Aggravated Assault                 32
11830         2005              58.01    Aggravated Assault                 13
3510          1998              83.00    Aggravated Assault                 13
1320          1997             107.00    Aggravated Assault                 40
8000          2002              46.00    Aggravated Assault                  4
11840         2005              58.02    Aggravated Assault                 14
7990          2002              45.00    Aggravated Assault                  5
3470          1998               8.00    Aggravated Assault                  0
11780         2005              53.02    Aggravated Assault                  2
8080          2002              53.01    Aggravated Assault                 57
11770         2005              53.01    Aggravated Assault                 72
8170          2002               6.00    Aggravated Assault                 31
11710         2005              48.00    Aggravated Assault                  6
...            ...                ...                   ...                ...
2049          1997              61.00  Violent Crimes Total                  6
10559         2004              53.02  Violent Crimes Total                  8
10569         2004              54.00  Violent Crimes Total                 20
10499         2004              49.00  Violent Crimes Total                 17
1859          1997              46.00  Violent Crimes Total                 10
10489         2004              48.00  Violent Crimes Total                 12
2089          1997              65.00  Violent Crimes Total                 10
10419         2004              41.00  Violent Crimes Total                  5
7059          2001               7.00  Violent Crimes Total                 12
10429         2004              42.00  Violent Crimes Total                 13
2120          1997              68.98  Violent Crimes Total                  8
10439         2004              43.00  Violent Crimes Total                 26
4049          1999              21.00  Violent Crimes Total                 13
659           1996              49.00  Violent Crimes Total                 18
7049          2001              69.00  Violent Crimes Total                  7
10449         2004              44.00  Violent Crimes Total                 38
2110          1997              67.98  Violent Crimes Total                 10
5789          2000              66.00  Violent Crimes Total                  9
4769          1999              85.00  Violent Crimes Tot

### Conclusions

From looking at the data, I have reached the conclusion that there a lot of crimes that happen in Seattle. It was surprising to also see the average amount of crimes comitted every year.


### Acknowledgments

I want to thank Ms. Sconyers for helping me on some of the code here. 